StarkInfra/StarkInfra/Utils/Parse.cs

```diff
@@ -1,74 +1,45 @@
-using System;
-using System.Linq;
-using System.Collections.Generic;
-using Newtonsoft.Json.Linq;
+﻿using System;
 using EllipticCurve;
-
+using StarkCore;
+using StarkCore.Utils;
 
 namespace StarkInfra.Utils
 {
-    internal static class Parse
+	public class Parse
     {
-        internal static SubResource ParseAndVerify(string content, string signature, string resourceName,
-            Api.ResourceMaker resourceMaker, User user, string key = null)
-        {
-            string verifiedContent = Verify(content, signature, user);
-            dynamic json = Utils.Json.Decode(verifiedContent);
-            if (key != null)
-            {
-                json = json[key]; 
-            }
-            return Api.FromApiJson(resourceMaker, json);
-        }
 
-        internal static string Verify(string content, string signature, User user)
-        {
-            Signature signatureObject;
-            try
-            {
-                signatureObject = Signature.fromBase64(signature);
-            } catch
-            {
-                throw new Error.InvalidSignatureError("The provided signature is not valid");
-            }
+        static string host = StarkHost.bank;
+        static string apiVersion = "v2";
+        static string sdkVersion = "0.2.0";
 
-            if (VerifySignature(content, signatureObject, user)) {
-                return content;
-            }
-            if (VerifySignature(content, signatureObject, user, true)) {
-                return content;
-            }
-
-            throw new Error.InvalidSignatureError("The provided signature and content do not match the Stark Infra public key");
-        }
-        
-        private static bool VerifySignature(string content, Signature signature, User user, bool refresh = false)
+        public static SubResource ParseAndVerify(string content, string signature, string resourceName, Api.ResourceMaker resourceMaker, User user, string key = null)
         {
-            
-            PublicKey publicKey = Utils.Cache.StarkInfraPublicKey;
 
-            if (publicKey is null || refresh)
-            {
-                publicKey = GetPublicKeyPem(user);
-            }
+            return StarkCore.Utils.Parse.ParseAndVerify(
```

Comment:
Aqui já que vc está importando StarkCore, não tem como deixar somente Parse.PaseAndVerify?

Assim como vc fez na linha 15 com Api.ResourceMaker

Reply:
nao rola, eu to fazendo um relay dessa função com o mesmo nome, se eu tirar a referencia ele vai apontar para a propria função